bentoml/_internal/io_descriptors/multipart.py

```diff
@@ -202,22 +211,61 @@ async def from_http_request(self, request: Request) -> dict[str, t.Any]:
         if ctype != b"multipart/form-data":
             raise BentoMLException(
                 f"{self.__class__.__name__} only accepts `multipart/form-data` as Content-Type header, got {ctype} instead."
-            )
+            ) from None
 
-        res: dict[str, t.Any] = dict()
         reqs = await populate_multipart_requests(request)
 
-        for k, i in self._inputs.items():
-            req = reqs[k]
-            v = await i.from_http_request(req)
-            res[k] = v
-        return res
+        return {
+            key: await io_.from_http_request(reqs[key])
+            for key, io_ in self._inputs.items()
+        }
 
     async def to_http_response(
         self, obj: dict[str, t.Any], ctx: Context | None = None
     ) -> Response:
-        res_mapping: dict[str, Response] = {}
-        for k, io_ in self._inputs.items():
-            data = obj[k]
-            res_mapping[k] = await io_.to_http_response(data, ctx)
+        res_mapping: dict[str, Response] = {
+            key: await io_.to_http_response(obj[key], ctx)
+            for key, io_ in self._inputs.items()
+        }
         return await concat_to_multipart_response(res_mapping, ctx)
+
+    async def from_proto(
+        self,
+        field: MessageMap[str, pb.Part],
+        *,
+        _use_internal_bytes_contents: bool = False,
+    ) -> dict[str, t.Any]:
+        if _use_internal_bytes_contents:
+            raise InvalidArgument(
+                f"cannot use '_internal_bytes_contents' with {self.__class__.__name__}"
+            ) from None
+        if len(set(field) - set(self._inputs)) != 0:
+            raise InvalidArgument(
+                f"'{self.__class__.__name__}' only accepts '{set(self._inputs)}' as input fields. Invalid fields are: {set(field) - set(self._inputs)}",
+            ) from None
+        return {
+            key: await self._inputs[key].from_proto(
+                input_pb, _use_internal_bytes_contents=_use_internal_bytes_contents
+            )
+            for key, input_pb in field.items()
+        }
+
+    async def to_proto(self, obj: dict[str, t.Any]) -> MessageMap[str, pb.Part]:
+        if len(set(obj) - set(self._inputs)) != 0:
+            raise InvalidArgument(
+                f"'{self.__class__.__name__}' only accepts '{set(self._inputs)}' as output fields. Invalid fields are: {set(obj) - set(self._inputs)}",
```

Comment:
I think this is confusing, because when rendered this will read something like:

`'Multipart' only accepts '{my_custom_field}' as output fields. Invalid fields are: {my_other_field}"

We probably should mention that this is based on user configuration somehow; maybe we include the arguments to the descriptor class? Also, we probably want to format the sets a little more nicely.

Reply:
I have updated the implementation if this makes more sense.